includes/class-products.php

```diff
@@ -95,103 +107,180 @@ public static function is_active() {
 		return class_exists( 'WooCommerce' ) && class_exists( 'WC_Subscriptions_Product' ) && defined( 'NEWSPACK_LISTINGS_SELF_SERVE_ENABLED' ) && NEWSPACK_LISTINGS_SELF_SERVE_ENABLED;
 	}
 
+	/**
+	 * Verify that the products associated with this feature set are valid and published.
+	 *
+	 * @return boolean True if valid.
+	 */
+	public static function validate_products() {
+		$valid    = true;
+		$products = self::get_products();
```

Comment:
When loading any admin page, I get a ton of 

```
PHP Notice:  Function wc_get_product was called <strong>incorrectly</strong>. wc_get_product should not be called before the woocommerce_init, woocommerce_after_register_taxonomy and woocommerce_after_register_post_type actions have finished
```

warnings, with stack trace pointing to this line.

Reply:
Hmm, I can't replicate for some reason—does 97c40cf fix the issue?

Reply:
It does 👌

Reply:
Not entirely – I still see the notice, with the following stack trace:

```
Newspack_Listings\Core::enable_listing_category_archives, 
Newspack_Listings\Settings::get_settings, 
Newspack_Listings\Settings::get_default_settings, 
Newspack_Listings\Products::validate_products, 
Newspack_Listings\Products::get_products
```

Which means the `get_products` method is called on `pre_get_posts` hook [here](https://github.com/Automattic/newspack-listings/blob/0dac9fb482b577de74e886900a7cf85eed2bd571/includes/class-core.php#L88).

Reply:
Ah, thanks for the stack trace. 12d641e should fix this by validating the products based on standard WP functions instead of relying on WC's `wc_get_products` function.